db/migrate/20170427011934_re_add_owner_to_application.rb

```diff
@@ -0,0 +1,7 @@
+class ReAddOwnerToApplication < ActiveRecord::Migration[5.0]
+  def change
+    add_column :oauth_applications, :owner_id, :integer, null: true
+    add_column :oauth_applications, :owner_type, :string, null: true
```

Comment:
Adding the foreign key is no problem.

I'm reluctant to remove the polymorphic association because it's part of `Doorkeeper` and the underlying code expects it, and removing it could have some weird side-effects. At the very least, removing `owner_type` breaks a bunch of specs I wrote which rely on [the Application fabricator](https://github.com/muffinista/mastodon/blob/application-prefs-section/spec/fabricators/application_fabricator.rb) and while I can probably figure out a way around that issue, it seems like it would introduce a lot of potential fragility.